src/datadog/span.cpp

```diff
@@ -40,6 +41,16 @@ Span::~Span() {
     data_->duration = now - data_->start;
   }
 
+#ifdef __linux__
+  // When a span is finished, we must update the span_id to its parent's.
+  if (elastic_apm_profiling_correlation_process_storage_v1 != nullptr &&
+      parent_id().has_value()) {
+    elastic_apm_profiling_correlation_tls_v1->valid = 0;
+    elastic_apm_profiling_correlation_tls_v1->span_id = parent_id().value();
+    elastic_apm_profiling_correlation_tls_v1->valid = 1;
```

Comment:
For future reference, if anyone's curious, I was reviewing this with @elsakeirouz and we noticed we're missing some synchronization here, [per the spec](https://github.com/elastic/apm/blob/bd5fa9c1/specs/agents/universal-profiling-integration.md#concurrency-safe-updates):

> Note that APM-agents must make sure that compilers do not reorder the steps listed above.